### DISCUSSION TITLE
Inject file content

### DISCUSSION THREAD

**Discussion by CrysK** [requarks/wiki#6576] [Help / Questions]
Hello,

is there a way that you can include the contents of a file in a (markdown) file?

An example:
In my page "coding" with markdown editor (but never mind), I have the lines:

```
''' php
=> /data/example.php
'''
```

Here in the code box the content of the uploaded "example.php" should be output. 
Is this possible?

Thanks a lot

**Comment by fnafox** [comment]
you could probably do that with html, i've never tried it but i see no reason why it shouldn't work

**Comment by CrysK** [comment]
@fnafox 
I am pretty sure it can't be done with html itself.

**Comment by CrysK** [comment]
With this, it is possible by JS:
``` html
<!DOCTYPE html>
<html>
<head>
<script>
window.onload = function() {
  loadContent();
};

function loadContent() {
  var filePath = document.getElementById("content").textContent;
  
  fetch(filePath)
	.then(response => response.text())
	.then(data => {
	  var content = "<pre v-pre='true' class='prismjs line-numbers'><code class='language-YAML'>" + data + "</code></pre>";
	  document.getElementById("content").innerHTML = content;
	})
	.catch(error => {
	  console.error('Error loading the file:', error);
	});
}
</script>
</head>
<body>
  <div id="content">/path/to/file.php</div>
</body>
</html>
```

The <script> can be executed only in "Body HTML injection". Nothing happens in the "Head". 

But the problem is that it was loaded after the js formatting. That's why the pasted text is not formatted anymore. 
How to insert the text before formatting? Or trigger the formatting again (via JS)?

**Reply by P4mes4n** [reply]
While your script does work as a head injection for me, I did not manage to solve the other problems.
Would actually be a cool feature.

**Comment by CrysK** [comment]
To be more precise, I miss this function of mkdocs:
https://squidfunk.github.io/mkdocs-material/reference/code-blocks/#embedding-external-files
**"Embedding external files"**
